release.py

```diff
@@ -0,0 +1,694 @@
+        logger.info(f'  {pr_url}')
+        if len(open_prs) > 1:
+            logger.info(
+                f'Note that there are {len(open_prs) - 1} more open update charm pins PRs.'
+            )
+
+
+def draft_release(owner: str, repo_name: str, base_branch: str):
+    """Create a draft release, update changelog, and create a PR for the release."""
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    check_update_charm_pins_prs(repo)
+
+    tag = get_new_tag_for_release(owner, repo, base_branch)
+    release = create_draft_release(repo, tag, base_branch)
+    if not release:
+        logger.error('Failed to create draft release.')
+        exit(1)
+    logger.info(f'Draft release created: {release.html_url}.')
+
+    categories, full_changelog = parse_release_notes(release.body)
+    notes = format_release_notes(categories, full_changelog)
+    print_release_notes(notes)
+
+    title, summary = input_title_and_summary(release)
+    if not title:
+        title = tag
+    notes = f'{summary}\n{notes}'
+
+    update_draft_release(release, title, notes)
+
+    changes = format_changes(categories, tag)
+    update_changes_file(changes, 'CHANGES.md')
+
+    update_versions_for_release(tag)
+
+    new_branch = f'release-prep-{tag}'
+    subprocess.run(['/usr/bin/git', 'checkout', '-b', new_branch], check=True)
+    changed_files = ['CHANGES.md', *[str(path) for path in VERSION_FILES.values()]]
+    for file in changed_files:
+        subprocess.run(['/usr/bin/git', 'add', file], check=True)
+    subprocess.run(['/usr/bin/git', 'commit', '-m', f'chore: prepare release {tag}'], check=True)
+    subprocess.run(['/usr/bin/git', 'push', 'origin', new_branch], check=True)
+    pr = repo.create_pull(
+        title=f'chore: update changelog and versions for {tag} release',
+        body=f'This PR prepares the release of version {tag}.',
+        head=f'{gh_client.get_user().login}:{new_branch}',  # "your_username:new_branch"
+        base=base_branch,
+    )
+    logger.info(f'Created PR: {pr.html_url}')
+
+
+def post_release(owner: str, repo_name: str, base_branch: str):
+    """Post-release actions: update version files and create a PR."""
+    new_branch = 'post-release'
+    local_branch = subprocess.run(
+        ['/usr/bin/git', 'branch', '--list', new_branch],
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip()
+    remote_branch = subprocess.run(
+        ['/usr/bin/git', 'ls-remote', '--heads', 'origin', new_branch],
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip()
+    exist = local_branch or remote_branch
+    if exist:
+        logger.error(
+            f'Branch "{new_branch}" already exists in the current repository. '
+            'Please double check and delete it first before post release'
+        )
+        exit(1)
+
+    subprocess.run(['/usr/bin/git', 'fetch', 'upstream'], check=True)
```

Comment:
These days my `canonical/operator` remote is called `origin` (and `benhoyt` is mine), not `upstream`. Can we avoid hard-coding this?

Could we do `git checkout {base_branch}` and `git pull` instead? And do we need to do the same "update to latest" before `draft_release` too?

Reply:
`base_branch` defaults to `main`, so `git checkout main` and `git pull` will get the forked repo's main, instead of `canonical/operator`'s main. I will make the remote name configurable.

Reply:
Before `draft_release`: I assume `tox -e draft-release` is executed in the forked ops repo, checked out at latest main, with no changes in the staging area. The reason I did not do any operations before `draft_release` is that in case you are working on something, it might ruin the result. I will add some description in HACKING.md.